Sources/UIKit/AdaptiveStyle.swift

```diff
@@ -10,37 +10,74 @@
 import UIKit
 
 /// A few default font scaling behaviors.
-public enum AdaptiveStyle {
+public struct AdaptiveStyle {
+
+    public enum Behavior {
+        case control
+        case body
+        case preferred
+
+        @available(iOS 11, tvOS 11, *)
+        case fontMetrics
+
+        case above(size: CGFloat, useFontNamed: String)
+        case below(size: CGFloat, useFontNamed: String)
+    }
+
+    public let behavior: Behavior
+
+    // These two values are used only in the fontMetrics case. They should be associated enum values, but marking enum
+    // cases as @available is not supported: https://github.com/apple/swift/pull/36327
+    public let textStyle: BonMotTextStyle?
+    public let maxPointSize: CGFloat?
+
+    init(behavior: Behavior, textStyle: BonMotTextStyle? = nil, maxPointSize: CGFloat? = nil) {
+        self.behavior = behavior
+        self.textStyle = textStyle
+        self.maxPointSize = maxPointSize
+    }
 
     /// Scale the font up or down based on the Dynamic Type slider, but do not
     /// grow into the Accessibility ranges.
-    case control
+    public static var control: AdaptiveStyle {
+        AdaptiveStyle(behavior: .control)
+    }
```

Comment:
Converted all these enum cases to static functions to maintain call site compatibility.